.github/workflows/ci.yml

```diff
@@ -0,0 +1,59 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build-and-test:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Restore tools
+        run: dotnet tool restore
+        working-directory: ./Src
+      - name: Build templates
+        # TargetFramework specified so this only runs once rather than per framework
+        run: dotnet msbuild -t:TextTemplateTransform -p:TargetFramework=NA
+        working-directory: ./Src
+      - name: Debug Build
+        run: dotnet build --configuration=Debug -p:TreatWarningsAsErrors=true
+        working-directory: ./Src
+      - name: Release Build
+        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true
+        working-directory: ./Src
+      - name: CPU Tests
+        run: dotnet test --configuration=Release
+        working-directory: ./Src/ILGPU.Tests.CPU
+      - name: Upload bin directory
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ runner.os }}-bin
```

Comment:
Is templating with `${{ runner.os }}` necessary?
I guess the idea is to have this job run on multiple platforms once possible, but we would only upload the Windows build, right?